tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -262,6 +288,60 @@ private int WriteTags(ref byte[] bytes, int offset, in SpanModel model, ITagProc
+                {
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasSiteNameTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasSiteNameBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasSiteKindTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasSiteKindBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasSiteTypeTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasSiteTypeBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasResourceGroupTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasResourceGroupBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasSubscriptionIdTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasSubscriptionIdBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasResourceIdTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasResourceIdBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasInstanceIdTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasInstanceIdBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasInstanceNameTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasInstanceNameBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasOperatingSystemTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasOperatingSystemBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasRuntimeTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasRuntimeBytes);
+                    count++;
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasExtensionVersionTagNameBytes);
+                    offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _aasExtensionVersionBytes);
+                }
+                else
+                {
+                    // the front-end identify AAS spans using aas.site.name and aas.site.type, so we need them on all spans
```

Comment:
Instead of having `_aasSiteNameBytes` and `_aasSiteTypeBytes` in both `if` branches, what do you think of having them outside the `if` so it's clear that they are always added? When I first read this, I thought they were only in the `else` clause.

Combined with my other comments, maybe something like this? (pseudo-code)
```
if (AAS)
{
    InitializeAasTags
	
	count += 2
	add "aas.site.name"
	add "aas.site.type"

	if (IsChunkOrphan)
	{
		count += 9 (?)
		add all the other aas tags
	}
}
```